doc/releases/changelog-dev.md

```diff
@@ -65,6 +65,9 @@
 
 <h3>Breaking changes 💔</h3>
 
+* `qml.is_commuting` no longer accepts the `wire_map` argument, which does not bring any functionality.
```

Comment:
I don't think we need to justify the change in the changelog @trbromley ?

Reply:
Given these types of things could irritate or confuse someone, it's good to justify the change being worth it.